swarm/storage/resource.go

```diff
@@ -804,3 +849,31 @@ func (self *ResourceHandler) keyDataHash(key Key, data []byte) common.Hash {
 	hasher.Write(data)
 	return common.BytesToHash(hasher.Sum(nil))
 }
+
+// if first byte is the start of a multihash this function will try to parse it
+// if successful it returns the length of multihash data, 0 otherwise
+func isMultihash(data []byte) int {
+	cursor := 0
+	hashtype, c := binary.Uvarint(data)
+	log.Trace("ismultihash", "hashtype", hashtype, "c", c)
+	if c <= 0 {
+		log.Debug("Corrupt multihash data, hashtype is unreadable")
+		return 0
+	}
+	cursor += c
+	hashlength, c := binary.Uvarint(data[cursor:])
+	log.Trace("ismultihash", "hashlength", hashlength, "c", c)
+	if c <= 0 {
+		log.Debug("Corrupt multihash data, hashlength is unreadable")
+		return 0
+	}
+	cursor += c
+	// we cheekily assume hashlength < maxint
+	inthashlength := int(hashlength)
+	log.Trace("ismultihash", "datalen", len(data), "hashlength", inthashlength, "cursor", c)
+	if len(data[cursor:]) < inthashlength {
+		log.Debug("Corrupt multihash data, hash does not align with data boundary")
+		return 0
+	}
+	return cursor + inthashlength
```

Comment:
Isn't this a little bit too much logging? They are fine during development, but probably they could be removed, or at least set to `Debug`. On the other hand the error cases are only logged with Debug level, although those could be Warns or Errors maybe.